tests/e2e/managedosversionchannel_test.go

```diff
@@ -134,5 +134,84 @@ var _ = Describe("ManagedOSVersionChannel e2e tests", func() {
 				Equal(""),
 			)
 		})
+
+		It("creates a list of ManagedOSVersion from a custom hook", func() {
+
+			versions := []provv1.ManagedOSVersion{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "foo"},
+					Spec: provv1.ManagedOSVersionSpec{
+						Version:    "v1",
+						Type:       "container",
+						MinVersion: "0.0.0",
+						Metadata: &fleet.GenericMap{
+							Data: map[string]interface{}{
+								"upgradeImage": "registry.com/repository/image:v1",
+							},
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "bar"},
+					Spec: provv1.ManagedOSVersionSpec{
+						Version:    "v2",
+						Type:       "container",
+						MinVersion: "0.0.0",
+						Metadata: &fleet.GenericMap{
+							Data: map[string]interface{}{
+								"upgradeImage": "registry.com/repository/image:v2",
+							},
+						},
+					},
+				},
+			}
+
+			b, err := json.Marshal(versions)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			By("Create a ManagedOSVersionChannel")
+			ui := catalog.NewManagedOSVersionChannel(
+				"testchannel2",
+				"custom",
+				map[string]interface{}{
+					"image":      "opensuse/tumbleweed",
+					"command":    []string{"/bin/bash", "-c", "--"},
+					"mountPath":  "/output",      // This defaults to /data
+					"outputFile": "/output/data", // This defaults to /data/output
+					"args":       []string{fmt.Sprintf("echo '%s' > /output/data", string(b))},
```

Comment:
It is unclear to me how this is ending into pod logs...

Reply:
the container here pushes the json into a file which is picked up later by a container running in that pod. The file is passed by to the main container of the pod which will sit and display the file content in the logs until the operator reaps it. 

I plan to move away from logs passing the JSON structure, but for the time being this minimally satisfies the logics so we can keep going and revisit this into its specific issue #24